tests/unit/Gibbon/LocaleTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+*/
+
+namespace Gibbon;
+
+use Gibbon\Contracts\Database\Connection;
+use Gibbon\Locale;
+use League\Container\Container;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers Locale
+ *
+ * Test against PO file from POT generated with this command:
+ *
+ * ```
+ * xgettext \
+ *   --language=PHP \
+ *   --from-code=UTF-8 \
+ *   --add-comments=L10N \
+ *   --keyword=translate:1 \
+ *   --keyword=__:1 \
+ *   -o mock/i18n/gibbon.pot \
+ *   LocaleTest.php
+ * ```
+ *
+ * PO File before translation:
+ * ```
+ * cp mock/i18n/gibbon.pot mock/i18n/tlh_KX/LC_MESSAGES/gibbon.po
+ * ```
+ *
+ * MO File
+ * ```
+ * msgfmt --check-header --check-domain -v \
+ *  -o mock/i18n/tlh_KX/LC_MESSAGES/gibbon.mo \
+ *  mock/i18n/tlh_KX/LC_MESSAGES/gibbon.po
+ * ```
+ */
+class LocaleTest extends TestCase
+{
+    private $mockPDO;
+    private $mockSession;
+
+    private $locale;
+    private $gibbonToRestore;
+
+    public function setUp()
+    {
+
+        // Setup the composer autoloader
+        $autoloader = require_once __DIR__.'/../../../vendor/autoload.php';
+
+        // Require the system-wide functions
+        require_once __DIR__.'/../../../functions.php';
+
+        // Create a stub for the Gibbon\session class
+        $this->mockSession = $this->createMock(session::class);
+        $this->mockSession
+            ->method('get')
+            ->willReturn(null); // always return null
+
+        // mocked locale object
+        $locale = new Locale(__DIR__ . '/mock', $this->mockSession);
+
+        $i18ncode = 'zh_TW';
+        putenv('LANGUAGE=');
```

Comment:
Hmm, this test didn't run successfully on my system until I removed these lines and used the Locale classes set locale method (which accounts for some variation in the way locales are set from system to system):

`$locale->setLocale('zh_TW');`

Reply:
You need to do `sudo locale-gen zh_TW.utf8` to make he test work on your system (assuming it is linux)

Reply:
Yep, he locale is absolutely there & properly configured. The setLocale method on the Locale class includes the variations for setting a locale across different OSs, which can differ slightly. If you replace your `putenv` and `setlocale` lines with the `$locale->setLocale('zh_TW');` it will account for this :)

Reply:
Using `Locale::setLocale` now. Seems OK.